update Zone (#1022)
vesoft-inc/nebula-docs: docs-2.0/4.deployment-and-installation/5.zone.md

> # Manage logical rack

Reviewer: Manage zone

> Nebula Graph supports the zone feature to manage Storage services in a cluster to achieve resource isolation.

Reviewer: resource isolation, aka, logical rack

## Background

!!! compatibility

    From Nebula Graph version 3.0, the Storage services added in the configuration files **CANNOT** be read or written directly. The configuration files only register the Storage services with the Meta service. You must run the `ADD HOSTS` command to read and write data to the Storage services.

Users can add the Storage services to a zone. Users specify a zone when creating a graph space, the graph space will be created in all the Storage services of the zone. Partitions and replicas are evenly stored in each zone. As shown in the figure below.

![Zone figure](zone.png)

> Six machines with Storage service are grouped in pairs and added to three zones. Specify these three zones to create graph space S1, the partitions and replicas are evenly stored in zone1~zone3.

Reviewer 2: ```suggestion
Six machines with Storage services are grouped in pairs and added to three zones. Specify these three zones to create graph space S1, the partitions and replicas are evenly stored in zone1~zone3.
```

## Scenarios

> - Expect the graph space created in some specified Storage services to achieve resource isolation.

Reviewer 2: ```suggestion
- Create a graph space in some specified Storage services to isolate resources.
```

> - Expect the cluster to perform a rolling upgrade. One or more servers need to be stopped and updated, and then put back into use until all the servers in the cluster are updated to the new version.

Reviewer 2: ```suggestion
- Perform cluster rolling upgrades. One or more servers need to be stopped and updated, and then start the servers until all the servers in the cluster are updated to the new version.
```

## Syntax

### ADD HOSTS...INTO NEW ZONE

Add the Storage services to a new zone.

```ngql
ADD HOSTS <ip>:<port> [,<ip>:<port> ...] [INTO NEW ZONE "<new_zone_name>"];
```

Example:

```ngql
nebula> ADD HOSTS 192.168.10.100:9779, 192.168.10.101:9779, 192.168.10.102:9779;
```

> If `INTO NEW ZONE "<new_zone_name>"` is not used, each Storage service is added to an independent zone that created automatically and name format is `default_zone_<host>_<port>`.

Reviewer 2: ```suggestion
If `INTO NEW ZONE "<new_zone_name>"` is not used, each Storage service is added to an independent zone created automatically, the name format of which is `default_zone_<host>_<port>`.
```

+------------------------------------+------------------+------+
```

### ADD HOSTS...INTO ZONE

Add the Storage services to an existing zone.

!!! note

    Use the [BALANCE DATA](../3.ngql-guide/18.operation-and-maintenance-statements/2.balance-syntax.md) command to balance the load after adding the Storage services.

```ngql
ADD HOSTS <ip>:<port> [,<ip>:<port> ...] INTO ZONE "<zone_name>";
```

### DROP HOSTS

Delete the Storage services from cluster.

!!! note

    You can not delete an in-use Storage service directly. You need to delete the associated graph space before deleting the Storage service.

```ngql
DROP HOSTS <ip>:<port> [,<ip>:<port> ...];
```

### SHOW ZONES

View all zones.

```ngql
SHOW ZONES;
```

### DESC ZONE

View a specified zone.

```ngql
DESCRIBE ZONE "<zone_name>";
DESC ZONE "<zone_name>";
```

### RENAME ZONE

Rename a zone.

```ngql
RENAME ZONE "<old_name>" TO "<new_name>";
```

### DROP ZONE

Delete a zone。

!!! note

    You can delete a zone only when there are no partitions in the zone.

```ngql
DROP ZONE "<zone_name>";
```

### MERGE ZONE...INTO

Merge Storage services in multiple zones into a new zone.

!!! note

>     The distribution of partitions will be checked during the merge to prevent different replica of the same partition from being distributed in the same zone.

Reviewer 2: ```suggestion
    The distribution of partitions will be checked during the merge to prevent different partition replicas from being distributed in the same zone.
```
